app/routers/files.py

```
@@ -1049,47 +1069,78 @@ def stream_file_ultra_optimized(path: Path):
+                        else:
+                            raise ValueError("Cannot decrypt random-key encrypted file without session key storage")
                         
-                        # Validate integrity if metadata available
-                        if metadata and 'original_hash' in metadata:
-                            import hashlib
-                            actual_hash = hashlib.sha256(decrypted_data).hexdigest()
-                            expected_hash = metadata['original_hash']
-                            if actual_hash != expected_hash:
-                                raise Exception(f"File integrity check failed! Expected: {expected_hash}, Got: {actual_hash}")
-                            print(f"OK: File integrity validated successfully")
+                        cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
```

Ruff (0.16.0)

[error] Undefined name `Cipher`

(F821)